pkg/streamingpromql/operators/selectors/extend_range_vector.go

```diff
@@ -0,0 +1,187 @@
+// anchored boundary points at the start and end of the specified range.
+//
+// A synthetic point is placed at both rangeStart and rangeEnd:
+//
+//   - Start value: taken from the last point with T <= rangeStart,
+//     or, if none exists, the first point with T > rangeStart.
+//   - End value:   taken from the first point with T >= rangeEnd,
+//     or, if none exists, the last point with T < rangeEnd.
+//
+// The provided view has already prepared the input range to include a point before the rangeStart,
+// and a point after the rangeEnd (smoothed only).
+//
+// Note that these points outside the original range are only provided if there is no existing point
+// already on the range boundary and the points are within a defined lookback/lookahead window.
+//
+// The returned slice includes all points within the range, along with the added/modified boundary points.
+//
+// This implementation is based on extendFloats() from promql/engine.go.
+func NewExtendedPointsForAnchored(view *types.FPointRingBufferView, rangeStart, rangeEnd int64, memoryConsumptionTracker *limiter.MemoryConsumptionTracker) (ExtendedPoints, error) {
+	extendedPoints, err := buildExtendedPoints(view, rangeStart, rangeEnd, memoryConsumptionTracker)
+	if err != nil {
+		return ExtendedPoints{}, err
+	}
+	return extendedPoints, nil
+}
+
+// NewExtendedPointsForSmoothed returns a slice of points adjusted to include
+// smoothed boundary points at the start and end of the specified range.
+//
+// As with NewExtendedPointsForAnchored, synthetic points are placed at both
+// rangeStart and rangeEnd. However, the boundary values are determined using
+// interpolation:
+//
+//   - Start value: interpolated between the last point with T < rangeStart and
+//     the first point with T > rangeStart, or taken directly from
+//     the first point with T >= rangeStart.
+//   - End value:   interpolated between the last point with T < rangeEnd and
+//     the first point with T >= rangeEnd, or taken directly from
+//     the last point with T <= rangeEnd.
+//
+// When an interpolated boundary point is used, an additional interpolated
+// value—applying counter-reset compensation—is written into the provided
+// smoothedHead or smoothedTail. If the returned range is later used by
+// rate() or increase(), these alternate compensated boundary points must be
+// substituted for the default smoothed ones.
+//
+// The returned slice includes all points within the specified range, along
+// with the added (interpolated or direct) boundary points.
+//
+// This implementation is based on extendFloats() from promql/engine.go.
+func NewExtendedPointsForSmoothed(view *types.FPointRingBufferView, rangeStart, rangeEnd int64, memoryConsumptionTracker *limiter.MemoryConsumptionTracker) (SmoothedPoints, error) {
+	smoothedPoints, err := buildExtendedSmoothedPoints(view, rangeStart, rangeEnd, memoryConsumptionTracker)
+	if err != nil {
+		return SmoothedPoints{}, err
+	}
+	return smoothedPoints, nil
```

Comment:
Given this is the only place that calls `buildExtendedSmoothedPoints`, why is the separate method needed?